### PR TITLE
Revert

### PR DESCRIPTION
**Pull request by Isafo**
most resent somewhat working function. Both correct functions are not functioning.

### CODE REVIEW

File: src/SPH.cpp [hunk 1]
```diff
@@ -75,32 +88,45 @@ void SPH::update(float dT)
...
+	for (int i = 0; i < D_NR_OF_PARTICLES; ++i) {
+		m_particles.dens[i] = 100.f;
+	}
+
 	update_density_and_factors(m_particles.mass, &m_particles.pos, m_particles.dens, scalar_values, m_neighbor_data, alpha, kernel_values);
 
+
 	// TODO: this function is incorretly, implemented correct
-	correct_divergence_error(k_v_i, scalar_values);
+	if(iter >= 1)
```
Comment (chren574) [inline src/SPH.cpp]: Wrong implementation of iteration (func. update())

File: src/SPH.cpp [hunk 2]
```diff
@@ -75,32 +88,45 @@ void SPH::update(float dT)
...
+	calculate_time_step(dT);
 
 	non_pressure_forces();
 
-	calculate_time_step(dT);
-
 	predict_velocities();
 
-	correct_density_error(alpha, dT, scalar_values, f_tot, k_v_i);
+	if(iter >= 2)
```
Comment (chren574) [inline src/SPH.cpp]: Wrong implementation of iteration (func. update())

File: src/SPH.cpp [hunk 3]
```diff
@@ -147,280 +174,262 @@ void SPH::find_neighborhoods() const
...
-	float sum_y{ 0.f };
-	float sum_z{ 0.f };
 	int neighbor_index;
-	float dx;
-	float dy;
-	float dz;
-	float kernel_gradient_x, kernel_gradient_y, kernel_gradient_z;
-
-	calculate_pressure_force(f_tot, k_v_i, &m_particles.pos, m_particles.mass, scalar_values, m_neighbor_data, m_particles.dens);
+	int iter = -1;
```
Comment (Liljan) [inline src/SPH.cpp]: Iter should start at 0.

File: src/SPH.cpp [hunk 4]
```diff
@@ -147,280 +174,262 @@ void SPH::find_neighborhoods() const
...
 
-		m_particles.vel.x[i] -= m_delta_t * sum_x;
-		m_particles.vel.y[i] -= m_delta_t * sum_y;
-		m_particles.vel.z[i] -= m_delta_t * sum_z;
-		sum_x = sum_z = sum_y = .0f;
-	}
+		// condition p_avg - p_0 > ny  ny = 1.01*(p_max_p_min) p_0 = 1000
+		pressure_avg = dens_avg / D_NR_OF_PARTICLES - C_REST_DENS;
+		ny = 1.01*(dens_max - dens_min);
+	} while (pressure_avg - C_REST_DENS > ny );
```
Comment (Liljan) [inline src/SPH.cpp]: This condition should check with correct number of iterations.

Comment (Liljan) [inline src/SPH.cpp]: Pressure should not be used at all. Density average is used instead.

File: src/SPH.cpp [hunk 5]
```diff
@@ -147,280 +174,262 @@ void SPH::find_neighborhoods() const
...
+	int ind;
 
-	neighbor_mass = mass;
 	for (auto particle = 0; particle < D_NR_OF_PARTICLES; ++particle)
 	{
 		nr_neighbors = neighbor_data[particle].n;
+
 		for (auto neighbor = 0; neighbor < nr_neighbors; ++neighbor)
 		{
 			neighbor_index = neighbor_data[particle].neighbor[neighbor];
```
Comment (Pelse862) [inline src/SPH.cpp]: //should add patcles own density as particle[particle] =1 * mass

**Pull request merged**